### PR TITLE
add the first version of the Alloy section

### PR DESCRIPTION
**Pull request by konnov**
This PR adds the first version of the report on the Alloy experiments

### CODE REVIEW

File: reports/spec4-alloy.tex [hunk 1]
```diff
@@ -1,7 +1,144 @@
...
+
+\begin{lstlisting}[language=alloy,columns=fullflexible]
+  pred noAccountableSafety {
+    disagreement and (3.mul[#slashableNodes] < #Signature)
+  }
+  run noAccountableSafety for 10 but 6 Block, 6 Checkpoint,
+                                     4 Signature, 6 FfgVote, 15 Vote, 5 int
+\end{lstlisting}
+
+Our goal was to see, how far Alloy could go, when we were increasing the scope.
```
Comment (thpani) [inline reports/spec4-alloy.tex:91]:
```suggestion
Thus it is interesting to increase these parameter values as much as possible to cover more scenarios, while keeping them small enough that the SAT solver still terminates within reasonable time.
```

File: reports/spec4-alloy.tex [hunk 2]
```diff
@@ -1,7 +1,144 @@
...
+Table~\ref{tab:alloy-mc} summarizes our experiments with Alloy when checking
+$\textit{noAccountableSafety}$ with Kissat. As we can see, we have managed to
+show accountable safety for the chain configurations of up to 6 blocks,
```
Comment (thpani) [inline reports/spec4-alloy.tex:94]:
```suggestion
show accountable safety for chain configurations of up to 6 blocks,
```

File: reports/spec4-alloy.tex [hunk 3]
```diff
@@ -1,7 +1,144 @@
...
+including the genesis. However, the most interesting configuration of 7 blocks
```
Comment (thpani) [inline reports/spec4-alloy.tex:95]:
```suggestion
including the genesis block. However, the most interesting configuration of 7 blocks
```

File: reports/spec4-alloy.tex [hunk 4]
```diff
@@ -1,7 +1,144 @@
...
+(on two chains) happens to be quite challenging for Alloy and Kissat.
```
Comment (thpani) [inline reports/spec4-alloy.tex:96]: Why "(on two chains)"?
Doesn't this spec allow for arbitrary block graphs?

File: reports/spec4-alloy.tex [hunk 5]
```diff
@@ -1,7 +1,144 @@
...
+        11 & 4 & 20 & 4 & 5 & 12 & 152s & 94M
+            \\ \midrule
+        12 & 5 & 25 & 4 & 5 & 12 & 234s & 117M
+            \\ \midrule
+        13 & 7 & 15 & 4 & 10 & 40 & over 10 days & 300 MB
+            \\ \bottomrule
+    \end{tabular}
+    \caption{Model checking experiments with Alloy and Kissat:
+      \textbf{blk} is the number of blocks, \textbf{chk} is the number
+      of checkpoints, \textbf{sig} is the number of validators' signatures,
```
Comment (thpani) [inline reports/spec4-alloy.tex:139]:
```suggestion
      of checkpoints, \textbf{sig} is the number of validator signatures,
```

File: reports/spec4-alloy.tex [hunk 6]
```diff
@@ -1,7 +1,144 @@
...
+      \textbf{ffg} is the number of FFG votes, \textbf{votes} is the number
+      of validators' votes.
```
Comment (thpani) [inline reports/spec4-alloy.tex:141]:
```suggestion
      of validator votes.
```

File: reports/spec4-alloy.tex (hunk 3 above)
Comment (thpani) [inline reports/spec4-alloy.tex:95]: Why is it "most interesting"?

File: reports/spec4-alloy.tex [hunk 7]
```diff
@@ -1,7 +1,144 @@
...
+            \\ \midrule
+        12 & 5 & 25 & 4 & 5 & 12 & 234s & 117M
+            \\ \midrule
+        13 & 7 & 15 & 4 & 10 & 40 & over 10 days & 300 MB
+            \\ \bottomrule
+    \end{tabular}
+    \caption{Model checking experiments with Alloy and Kissat:
+      \textbf{blk} is the number of blocks, \textbf{chk} is the number
+      of checkpoints, \textbf{sig} is the number of validators' signatures,
+      \textbf{ffg} is the number of FFG votes, \textbf{votes} is the number
```
Comment (thpani) [inline reports/spec4-alloy.tex:140]: Maybe add a footnote that `ffg` bounds the number of FFG votes we would see in `votes`?

**Review by thpani — COMMENTED** [review]
Good exposition!

Comments below, I'll commit the typo/wording ones and leave the rest for you to resolve.

File: reports/spec4-alloy.tex (hunk 4 above)
Comment (konnov) [reply reports/spec4-alloy.tex:96]: True, this spec has an arbitrary graph

File: reports/spec4-alloy.tex [hunk 8]
```diff
@@ -1,7 +1,144 @@
...
+    slot: Int
+  }
+
+  one sig GenesisPayload extends Payload {}
+  one sig GenesisBlock extends Block {}
+\end{lstlisting}
+
+In contrast to~\texttt{Spec 4}, our Alloy specification does not describe a
+state machine, but it specifies an arbitrary single state of the protocol. As
+in~\texttt{Spec 4}, we employ the same trick to restrict the justified
```
[the remark this answers is not in the document]
Comment (konnov) [reply reports/spec4-alloy.tex:52]: rephrased

File: reports/spec4-alloy.tex (hunk 3 above)
Comment (konnov) [reply reports/spec4-alloy.tex]: added an explanation

**Review by thpani — APPROVED** [review]

**Pull request merged**